src/main/java/org/springframework/integration/kafka/listener/QueueingMessageListenerInvoker.java

```diff
@@ -94,63 +105,69 @@ public void enqueue(KafkaMessage message) {
 	}
 
 	public void start() {
+
+		if(ringBufferProcessor != null){
+			stop(-1l);
```

Comment:
Frankly: I don't like this code. `start()` must be *idempotent*. And only the upstream `stop()` should reset the state. So, let me revert this dangerous code!